src/layout/constraint.rs

```diff
@@ -364,6 +366,18 @@ impl Constraint {
             .map(Constraint::Proportional)
             .collect_vec()
     }
+
+    pub fn rank(&self) -> usize {
+        match self {
+            Constraint::Fixed(_) => 0,
+            Constraint::Min(_) => 1,
+            Constraint::Max(_) => 2,
+            Constraint::Length(_) => 3,
+            Constraint::Percentage(_) => 4,
+            Constraint::Ratio(_, _) => 5,
+            Constraint::Proportional(_) => 6,
+        }
+    }
```

Comment:
I'm not able to change it to `self as usize`, presumably because there's a payload attached to the enum variant? I'm getting an error that implies that I can only convert from a primitive type when I use `self as usize`.

I added a test for this.

Reply:
Got it. I didn't realize that, but it makes sense.